src/compiler/transformers/esnext.ts

```diff
@@ -917,6 +1172,28 @@ namespace ts {
         );
     }
 
+    const classPrivateFieldGetHelper: EmitHelper = {
+        name: "typescript:classPrivateFieldGet",
+        scoped: false,
+        text: `var _classPrivateFieldGet = function (receiver, privateMap) { if (!privateMap.has(receiver)) { throw new TypeError("attempted to get private field on non-instance"); } return privateMap.get(receiver); };`
+    };
+
+    function createClassPrivateFieldGetHelper(context: TransformationContext, receiver: Expression, privateField: Identifier) {
+        context.requestEmitHelper(classPrivateFieldGetHelper);
+        return createCall(getHelperName("_classPrivateFieldGet"), /* typeArguments */ undefined, [ receiver, privateField ]);

```

Comment:
`privateField` here is an AST Node (Identifier) that will be reused in multiple parts of the AST. Is that safe? It's conceivable that a later stage in the transform pipeline might modify one branch of the AST and not expect another branch to change as a result.

Reply:
I think the way you did it is how it's done elsewhere, too, but I'm curious about how it works.